server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/LongValuesSource.java

```diff
@@ -38,45 +38,81 @@
 import org.elasticsearch.search.aggregations.LeafBucketCollector;
 
 import java.io.IOException;
+import java.util.function.LongConsumer;
 import java.util.function.LongUnaryOperator;
 import java.util.function.ToLongFunction;
 
 /**
  * A {@link SingleDimensionValuesSource} for longs.
  */
 class LongValuesSource extends SingleDimensionValuesSource<Long> {
+    private final BigArrays bigArrays;
     private final CheckedFunction<LeafReaderContext, SortedNumericDocValues, IOException> docValuesFunc;
     private final LongUnaryOperator rounding;
 
-    private final LongArray values;
+    private BitArray bits;
+    private LongArray values;
     private long currentValue;
+    private boolean missingCurrentValue;
 
-    LongValuesSource(BigArrays bigArrays, MappedFieldType fieldType,
-                     CheckedFunction<LeafReaderContext, SortedNumericDocValues, IOException> docValuesFunc,
-                     LongUnaryOperator rounding, DocValueFormat format, Object missing, int size, int reverseMul) {
-        super(format, fieldType, missing, size, reverseMul);
+    LongValuesSource(BigArrays bigArrays, LongConsumer breakerConsumer,
```

Comment:
It looks like `breakerConsumer` isn't used here, but is used in `DoubleValuesSource`.  Was that intentional?